eox_tenant/tenant_wise/proxies.py

```diff
@@ -126,43 +158,98 @@ def get_value_for_org(cls, org, val_name, default=None):
+            cls.set_key_to_cache(cache_key, result)
+            return result
+
+        result = Microsite.get_value_for_org(org, val_name)
+        result = result if result else default
+        cls.set_key_to_cache(cache_key, result)
+
+        return result
+
+    @classmethod
+    def __deprecated_get_value_for_org(cls, org, val_name, default=None):
+        """
+        This method iterates over all the TenantConfig and Microsite objects, set the cache for the given
+        val_name and all organizations and returns the value for the given org.
+
+        DEPRECATED: use __get_value_for_org method instead.
+        """
+        warnings.warn(
+            "__deprecated_get_value_for_org, MySQL 5.7+ is required to use __get_value_for_org instead.",
+            DeprecationWarning
+        )
+
+        if not cls.has_configuration_set():
+            return default
+
+        cache_key = "org-value-{}-{}".format(org, val_name)
+        cached_value = cache.get(cache_key)
+
+        if cached_value:
+            return cached_value
+
+        result = None
+        tenant_config = TenantConfig.objects.values_list("lms_configs", flat=True)
+        microsite_config = Microsite.objects.values_list("values", flat=True)  # pylint: disable=no-member
 
         for config in chain(tenant_config, microsite_config):
             try:
-                current = json.loads(config[0])
-                org_filter = current.get("course_org_filter", {})
-            except IndexError:
+                if isinstance(config, six.string_types):
+                    config = json.loads(config)
+
+                org_filter = config.get("course_org_filter", {})
+            except AttributeError:
                 continue
 
             if org_filter:
                 if isinstance(org_filter, six.string_types):
                     org_filter = set([org_filter])
-                if org in org_filter:
-                    result = current.get(val_name, default)
-                    cls.set_key_to_cache(cache_key, result)
-                    return result
 
-        return default
+                for organization in org_filter:
+                    if org == organization and not result:
+                        result = config.get(val_name, default)
 
-    @classmethod
-    def has_configuration_set(cls):
-        """
-        We always require a configuration to function, so we can skip the query
-        """
-        return getattr(settings, "USE_EOX_TENANT", False)
+                    key = "org-value-{}-{}".format(organization, val_name)
+                    cls.set_key_to_cache(key, config.get(val_name, default))
+
+        if not result:
+            logger.warning("Tha value for %s and org %s has not been found", val_name, org)
+            cls.set_key_to_cache(cache_key, default)
+            result = default
+
+        return result
 
     @classmethod
-    def set_key_to_cache(cls, key, value):
+    def pre_load_values_by_org(cls, val_name):
```

Comment:
When will this function be called?

Reply:
For now we have 2 candidates.
1) when a superuser loads the home page of studio. We will do this directly from the platform, to address the fact that superusers try to load about 6k courses and thus do 12k calls to the get value by org.

2) when a superuser calls the schedule and details of a course in studio. We have noticed that this page was also speed up when we preloaded the cache.

Since calling the funcion has a high cost, we might need to check the cache first to see if it is actually necessary to do so. But that should be done also in there.
@andrey-canon we need to be aware of this when we do the PR to start calling this.

Reply:
Does it make sense to add this function to this PR? Right now I think that it is not being used

Reply:
Yes of course. As soon as this is merged we will be merging as well the PR that uses it on edx-platform. Idk if the PR exists already, but we are already working on it.